tests/upgrades/test_bookmarks.py

```diff
@@ -0,0 +1,96 @@
+"""Test for bookmark related Upgrade Scenario's
+
+:Requirement: Upgraded Satellite
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: API
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from nailgun import entities
+from upgrade_tests import post_upgrade
+from upgrade_tests import pre_upgrade
+
+from robottelo.constants import BOOKMARK_ENTITIES
+from robottelo.test import APITestCase
+
+
+BOOKMARK_ENTITIES = list(BOOKMARK_ENTITIES)
+
+
+class ScenarioPositiveBookmark(APITestCase):
+    """
+    Created Bookmarks in pre upgrade should be unchanged after post upgrade
+    and verify the create and delete operation.
+
+    Test Steps:
+
+        1. Before Satellite upgrade:
+        2. Create set of bookmarks using available bookmark entities
+        3. Check the bookmark controller name and name after creation
+        7. Upgrade satellite.
+        8. Check the bookmarks status after upgrade.
+        9. Delete the bookmarks
+
+    :BZ: 1833264, 1826734
+
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        cls.bookmark_postfix = "_pre_upgrade_bookmark"
+
+    @pre_upgrade
+    def test_pre_create_bookmark(self):
+        """Create a bookmark for set of system entities
+
+        :id: c4f90034-ea57-4a4d-9b73-0f57f824d89e
+
+        :Steps:
+
+            1. Create a bookmark before the upgrade using fixed name and fixed controller
+            2. Check the book mark status
+
+        :expectedresults: Bookmark should be created succesfully.
+
+        :CaseImportance: Critical
+        """
+
+        for entity in BOOKMARK_ENTITIES:
+            book_mark_name = entity["name"] + self.bookmark_postfix
+            bm = entities.Bookmark(
+                controller=entity['controller'], name=book_mark_name, public=False
+            ).create()
+            assert bm.controller == entity['controller']
+            assert bm.name == book_mark_name
+
+    @post_upgrade(depend_on=test_pre_create_bookmark)
+    def test_post_create_bookmark(self):
+        """Check the status of bookmark for all system entities after upgrade
+
+        :id: 3b3abb85-cad2-4cbb-ad21-2780523351fd
+
+        :Steps:
+
+            1. Check the book mark status after post upgrade
+            2. Remove the book mark
+
+        :expectedresults: Bookmark details should be unchanged after upgrade
+
+        :CaseImportance: Critical
+        """
+        for entity in BOOKMARK_ENTITIES:
+            book_mark_name = entity["name"] + self.bookmark_postfix
+            bm = entities.Bookmark().search(query={'search': 'name="{0}"'.format(book_mark_name)})[
+                0
+            ]
+            assert bm.name == book_mark_name
+            assert bm.controller == entity['controller']
+            bm.delete()
```

Comment:
Do you think we need to add the update as well an extra level of test step from our side? I know it does not exist in their in Bugzilla. It up to you non-blocking comment from my side

Reply:
Thanks, Omkar for your valuable input, I have added one more scenario for the public enabled bookmark, This will help us to verify all attributes of the bookmark.